### DISCUSSION TITLE
How to implement PaginateWithBrowser

### DISCUSSION THREAD

**Discussion by melanki** [alex-on-ai/WebReaper#22] [Q&A]
Hi @pavlovtech ,

 I would like assistance with a usecase I am working on, as I find the current documentation a bit lacking.

Example case: https://www.farco.no/merke/victron
This page uses Javascript pagination(?), as such we must press the correct page or next page button and wait for the page to re-render.

**Comment by alex-on-ai** [comment]
Hi @melanki,

Good point, I will cover this use case in the documentation.

For now, give a try to the following method:

```
.PaginateWithBrowser(".linkSelector", ".paginationSelector") // use instead of .Paginate
```

Additionally, this [documentation section](https://github.com/pavlovtech/WebReaper#parsing-single-page-applications) might be useful.

**Comment by alex-on-ai** [comment]
Still the same approach in the current version (5.x): use `.PaginateWithBrowser("a.itemLinkSelector", ".nextPageSelector")` instead of `.Paginate(...)` when the pagination is JavaScript-driven — it drives the next-page control in a headless browser and waits for the re-render.

v5 unified the HTTP and browser page loaders behind a single `IPageLoader` (`docs/adr/0004-one-page-loader-transport-seam.md`), but `PaginateWithBrowser` behaviour is unchanged.

If the "next" control only mutates the DOM with no navigation at all, combine a browser fetch with an explicit interaction instead:

```csharp
.GetWithBrowser(new[] { url }, a => a.Click(".next").WaitForNetworkIdle())
.Follow(".itemLink")
```
